Lib/fontParts/base/font.py

```diff
@@ -1583,116 +2176,180 @@ def _getCharacterMapping(self):
+            the :class:`BaseLayer` instances to select.
+        :return: A :class:`tuple` of layer names representing the currently
+            selected :class:`BaseLayer` instances.
+
+        Getting selected layer names::
 
             >>> for name in layer.selectedLayerNames:
             ...     print(name)
 
         Setting selected layer names:
 
             >>> layer.selectedLayerNames = ["A", "B", "C"]
+
         """
     )
 
-    def _get_base_selectedLayerNames(self):
+    def _get_base_selectedLayerNames(self) -> Tuple[str, ...]:
         selected = tuple([normalizers.normalizeLayerName(name) for
                           name in self._get_selectedLayerNames()])
         return selected
 
-    def _get_selectedLayerNames(self):
-        """
-        Subclasses may override this method.
+    def _get_selectedLayerNames(self) -> Tuple[str, ...]:
+        """Get the selected glyph layer names in the native font layer.
+
+        This is the environment implementation of
+        the :attr:`BaseFont.selectedLayerNames` property getter.
+
+        :return: A :class:`tuple` of layer names representing the currently
+            selected :class:`BaseLayer` instances.
+
+        .. note::
+
+            Subclasses may override this method.
+
         """
-        selected = [layer.name for layer in self.selectedLayers]
-        return selected
+        return tuple(layer.name for layer in self.selectedLayers)
 
-    def _set_base_selectedLayerNames(self, value):
+    def _set_base_selectedLayerNames(self, value: List[str]) -> None:
         normalized = [normalizers.normalizeLayerName(name) for name in value]
         self._set_selectedLayerNames(normalized)
 
-    def _set_selectedLayerNames(self, value):
-        """
-        Subclasses may override this method.
+    def _set_selectedLayerNames(self, value: List[str]) -> None:
+        """Set the selected glyph layer names in the native font layer.
+
+        This is the environment implementation of
+        the :attr:`BaseFont.selectedLayerNames` property setter.
+
+        :param value: The :class:`list` of layer names representing
+            the :class:`BaseLayer` instances to select.
+
+        .. note::
+
+            Subclasses may override this method.
+
         """
         select = [self.layers(name) for name in value]
         self.selectedLayers = select
 
     # guidelines
 
-    selectedGuidelines = dynamicProperty(
+    selectedGuidelines: dynamicProperty = dynamicProperty(
         "base_selectedGuidelines",
-        """
-        A list of guidelines selected in the font.
+        """Get or set the selected guidelines in the font.
+
+        :param value: The :class:`list` of :class:`BaseGuideline` instances
+            to select.
+        :return: A :class:`tuple` of currently selected :class:`BaseGuideline`
+            instances.
 
-        Getting selected guideline objects:
+        Getting selected guideline objects::
```

Comment:
I do like this breaking up. Does it need an overall `Example` heading? If so, this would apply in other places here